web/src/lib/domain-logic/getXrayRenewalEvent.ts

```diff
@@ -0,0 +1,25 @@
+import { XrayRegistrationCalendarEvent } from "@businessnjgovnavigator/shared/calendarEvent";
+import { XrayData } from "@businessnjgovnavigator/shared/xray";
+import dayjs from "dayjs";
+
+export const getXrayRenewalEvent = (
+  xrayRegistrationData: XrayData | undefined,
+  year: number,
+  month?: number,
+): XrayRegistrationCalendarEvent | undefined => {
+  if (xrayRegistrationData === undefined) return;
+
+  if (dayjs(xrayRegistrationData.expirationDate).year() !== year) return;
+
+  const isMonthDefined = month !== undefined;
+  if (isMonthDefined && dayjs(xrayRegistrationData.expirationDate).month() !== month) return;
+
+  const isActiveOrExpired =
+    xrayRegistrationData.status === "ACTIVE" || xrayRegistrationData.status === "EXPIRED";
```

Comment:
How would an expired event show on the calendar? I believe the way the logic is written, events in the past will not show.